### PR TITLE
Enhancement: Extract UpdatePasswordAction from ForgotController

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] extracts an `UpdatePasswordAction` from the `ForgotController`

Follows #912.

### CODE REVIEW

File: classes/Http/Action/Forgot/UpdatePasswordAction.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        if (empty($resetCode)) {
+            throw new \Exception();
+        }
+
+        try {
+            $user = $this->accountManagement->findById($userId);
+        } catch (\RuntimeException $e) {
+            echo $e;
+            die();
```
Comment (localheinz) [inline classes/Http/Action/Forgot/UpdatePasswordAction.php]: This case is currently untested, hehe.

:trollface: 

I suggest we remove it soon.

**Comment by BackEndTea** [comment]
Shouldn't this also delete the ```updatePasswordAction``` in the ```ForgotController```?

**Comment by localheinz** [comment]
Ha, forgot to remove it, thank you for catching it, @BackEndTea!

**Comment by localheinz** [comment]
Thank you, @BackEndTea!

**Pull request merged**